extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -0,0 +1,128 @@
+     *
+     * @return
+     */
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FeatureBuildItem.SWAGGER_UI);
+    }
+
+    /**
+     * Register the Swagger UI servlet extension
+     *
+     * @param template - Swagger UI runtime template
+     * @param container - the BeanContainer for creating CDI beans
+     * @return servlet extension build item
+     */
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    public void registerSwaggerUiServletExtension(SwaggerUiTemplate template,
+            BuildProducer<ServletExtensionBuildItem> servletExtension,
+            BeanContainerBuildItem container,
+            ShutdownContextBuildItem shutdown) {
+        if (launch.getLaunchMode().isDevOrTest()) {
+            try {
+                ResolvedArtifact artifact = getSwaggerUiArtifact();
+                Path tempDir = Files.createTempDirectory(TEMP_DIR_PREFIX);
+                extractSwaggerUi(artifact, tempDir);
+                updateApiUrl(tempDir.resolve("index.html"));
+                servletExtension.produce(
+                        new ServletExtensionBuildItem(
+                                template.createSwaggerUiExtension(
+                                        swaggerUiConfig.path,
+                                        tempDir.toAbsolutePath().toString(),
+                                        container.getValue(),
+                                        shutdown)));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private ResolvedArtifact getSwaggerUiArtifact() {
+        ClassPathArtifactResolver resolver = new ClassPathArtifactResolver(SwaggerUiProcessor.class.getClassLoader());
+        return resolver.getArtifact(SWAGGER_UI_WEBJAR_GROUP_ID, SWAGGER_UI_WEBJAR_ARTIFACT_ID, null);
+    }
+
+    private void extractSwaggerUi(ResolvedArtifact artifact, Path resourceDir) throws IOException {
+        File artifactFile = artifact.getArtifactPath().toFile();
+        JarFile jarFile = new JarFile(artifactFile);
+        Enumeration<JarEntry> entries = jarFile.entries();
+        String versionedSwaggerUiWebjarPrefix = format("%s/%s/", SWAGGER_UI_WEBJAR_PREFIX, artifact.getVersion());
+        while (entries.hasMoreElements()) {
+            JarEntry entry = entries.nextElement();
+            if (entry.getName().startsWith(versionedSwaggerUiWebjarPrefix) && !entry.isDirectory()) {
+                InputStream inputStream = jarFile.getInputStream(entry);
+                String filename = entry.getName().replace(versionedSwaggerUiWebjarPrefix, "");
+                Files.copy(inputStream, resourceDir.resolve(filename));
+            }
+        }
+    }
+
+    private void updateApiUrl(Path indexHtml) throws IOException {
+        String content = new String(Files.readAllBytes(indexHtml));
+        content = content.replaceAll(SWAGGER_UI_DEFAULT_API_URL, OPEN_API_DEFAULT_URL);
```

Comment:
If this has been modified by config shouldn't this be taken into account? Also if you make both these changes then you would need to also cache this path, so if it changes you would need to update the indexHtml accordingly (or just delete and re-run).